inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -0,0 +1,438 @@
+    std::fill_n(dstData, offset, padValue);
+
+    parallel_nt(0, [&](const int ithr, const int nthr) {
+        size_t start = 0, end = 0;
+        SizeVector counters(dimsSize_1, 0);
+        splitter(workAmountSrc, nthr, ithr, start, end);
+        SizeVector countersNext(dimsSize_1, 0);
+
+        parallel_init(start, dimsSize_1, counters, srcDims);
+        parallel_init(start, dimsSize_1, countersNext, srcDims);
+        parallel_step(dimsSize_1, countersNext, srcDims);
+        int srcIdx = 0;
+        int dstIdx = padsBegin[dimsSize_1];
+        int dstIdxNext = padsBegin[dimsSize_1];
+        for (size_t i = 0; i < dimsSize_1; ++i) {
+            srcIdx += counters[i] * params.srcStrides[i];
+            dstIdx += (padsBegin[i] + counters[i]) * params.dstStrides[i];
+            dstIdxNext += (padsBegin[i] + countersNext[i]) * params.dstStrides[i];
+        }
+        if (dstIdxNext <= dstIdx) dstIdxNext = params.dstStrides[0] * dstDims[0];
+
+        for (size_t iwork = start; iwork < end; ++iwork, srcIdx += inputSV) {
+            cpu_memcpy(&dstData[dstIdx], &srcData[srcIdx], sizeof(float) * inputSV);
+            std::fill_n(&dstData[dstIdx + inputSV], dstIdxNext - dstIdx - inputSV, padValue);
+
+            for (int j = dimsSize_1 - 1; j >= 0; j--) {
+                counters[j] = (counters[j] + 1) % srcDims[j];
+                if (counters[j] != 0) {
+                    dstIdx += params.dstStrides[j];
+                    break;
+                } else {
+                    dstIdx = padsBegin[dimsSize_1];
+                    for (size_t i = 0; i < dimsSize_1; ++i)
+                        dstIdx += (padsBegin[i] + counters[i]) * params.dstStrides[i];
+                }
+            }
+
+            for (int j = dimsSize_1 - 1; j >= 0; j--) {
+                countersNext[j] = (countersNext[j] + 1) % srcDims[j];
+                if (countersNext[j] != 0) {
+                    dstIdxNext += params.dstStrides[j];
+                    break;
+                } else {
+                    dstIdxNext = padsBegin[dimsSize_1];
+                    for (size_t i = 0; i < dimsSize_1; ++i)
+                        dstIdxNext += (padsBegin[i] + countersNext[i]) * params.dstStrides[i];
+                }
+            }
+            if (dstIdxNext <= dstIdx) dstIdxNext = params.dstStrides[0] * dstDims[0];
+        }
+    });
+}
+
+void MKLDNNPadNode::padConstantOptimalImpl(const float *srcData, float* dstData, SizeVector srcDims) {
+    size_t outer_size = srcDims[0] * srcDims[1];
+    uint8_t sizeData = getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].desc.getPrecision().size();
+
+    parallel_for2d(srcDims[2], outer_size, [&](size_t i, size_t j) {
+        auto dstPtr = dstData + i * params.dstStrides[2] + j * params.dstStrides[1];
+        auto srcPtr = srcData + i * params.srcStrides[2] + j * params.srcStrides[1];
+
+        cpu_memcpy(dstPtr, srcPtr, srcDims[3] * sizeData);
```

Comment:
This function accepts pointers to floats. Maybe it's better to replace `sizeData` with `sizeof(float)` and remove `sizeData` above? Or `uint8_t sizeData = sizeof(float)`